src/org/python/core/PrePy.java

```diff
@@ -417,6 +417,8 @@ public static String getJarFileNameFromURL(URL url) {
                         // path contains the target class in a JAR (named in group 1).
                         // Make a file URL from all the text up to the end of group 1.
                         fileURI = new URL("file:" + path.substring(0, m.end(1))).toURI();
+                        // When URL is Relative,convert to Absolute path
+                        fileURI = new URI("file:"+(new File(fileURI.toString().substring(5))).getCanonicalPath());
```

Comment:
Gosh, is it really that simple? Thanks for finding it.

This is going a long way only to come back and start again. We ought to build `fileURI` only once. Can you try something like:
```
File jarFile = new File(path.substring(0, m.end(1)));
fileURI = new URI("file:"+jarFile.getCanonicalPath())
```

Secondly, this bug is present in 2.7.2 and not in 2.7.1 (probably) because I refactored some code and the tests didn't catch my mistake. Is there something to be added here:
https://github.com/jython/jython/blob/63359da881edcf549909ee0492edcc91bb1acf6e/tests/java/org/python/core/PySystemStateTest.java#L114
 that would fail without your change, and pass with it. In other words, can we have a test that makes sure no-one undoes your good work? (I think that's the right place to exercise the fix.)